kv/union_store.go

```diff
@@ -39,6 +39,22 @@ type UnionStore interface {
 	GetMemBuffer() MemBuffer
 }
 
+// AssertionType is the type of a assertion.
+type AssertionType int
+
+// The AssertionType constants.
+const (
+	NoCheck AssertionType = iota
+	MustExist
+	MustNotExist
+)
+
+// SafeStore is store that support assertions.
+type SafeStore interface {
```

Comment:
Can we just add `SetAssertion` in the `Transaction` interface?

Reply:
It's not a good choice. `Assertion` is optional, and has nothing to do with `Transaction`.